os/fs/smartfs/smartfs_utils.c

```diff
@@ -1163,8 +1164,8 @@ int smartfs_finddirentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *d
 				/* Read the next directory in the chain */
 
 				smartfs_setbuffer(&readwrite, dirsector, 0, fs->fs_llformat.availbytes, (uint8_t *)fs->fs_rwbuffer);
-				ret = FS_IOCTL(fs, BIOC_READSECT, (unsigned long)&readwrite);
-				if (ret < 0) {
+				tmp = FS_IOCTL(fs, BIOC_READSECT, (unsigned long)&readwrite);
+				if (tmp < 0) {
```

Comment:
what will be the "ret" value in case of errout? we need to return proper error ret value.

Reply:
Updated for proper return values